redfish-core/lib/account_service.hpp

```diff
@@ -807,6 +809,81 @@ inline void
         ldapConfigInterface, "UserNameAttribute",
         std::variant<std::string>(userNameAttribute));
 }
+
+inline void acfGetMessageSetProperties(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::tuple<std::vector<uint8_t>, bool, std::string>& messageFDbus)
+{
+    asyncResp->res.jsonValue["Oem"]["IBM"]["@odata.type"] =
+        "#OemManagerAccount.v1_0_0.IBM";
+    asyncResp->res.jsonValue["Oem"]["IBM"]["ACF"]["@odata.type"] =
+        "#OemManagerAccount.v1_0_0.ACF";
+    // Get messages from call to InstallACF and add values to json
+    std::vector<uint8_t> ACFFile = std::get<0>(messageFDbus);
+    std::string decodeACFFile(ACFFile.begin(), ACFFile.end());
+    std::string encodedACFFile = crow::utility::base64encode(decodeACFFile);
+
+    bool ACFInstalled = std::get<1>(messageFDbus);
+    std::string expirationDate = std::get<2>(messageFDbus);
+
+    asyncResp->res
+        .jsonValue["Oem"]["IBM"]["ACF"]["WarningLongDatedExpiration"] = nullptr;
+    asyncResp->res.jsonValue["Oem"]["IBM"]["ACF"]["ACFFile"] = nullptr;
+    asyncResp->res.jsonValue["Oem"]["IBM"]["ACF"]["ExpirationDate"] = nullptr;
+
+    if (ACFInstalled)
+    {
+        asyncResp->res.jsonValue["Oem"]["IBM"]["ACF"]["ExpirationDate"] =
+            expirationDate;
+
+        asyncResp->res.jsonValue["Oem"]["IBM"]["ACF"]["ACFFile"] =
+            encodedACFFile;
+
+        std::time_t result = std::time(nullptr);
+        // YYYY-MM-DD format
+
+        // Parse expirationDate to get difference between now and expiration
+
+        std::string expirationDateCpy = expirationDate;
+        std::string delimiter = "-";
+        std::vector<int> parseTime;
+
+        size_t pos = 0;
+        std::string token;
+        while ((pos = expirationDateCpy.find(delimiter)) != std::string::npos)
+        {
+            token = expirationDateCpy.substr(0, pos);
+            parseTime.push_back(std::strtol(token.c_str(), nullptr, 10));
+            expirationDateCpy.erase(0, pos + delimiter.length());
+        }
+        parseTime.push_back(
+            std::strtol(expirationDateCpy.c_str(), nullptr, 10));
+        std::tm tm{}; // zero initialise
+        tm.tm_year = parseTime.at(0) - 1900;
```

Comment:
What format is this time in? 
Could you add a comment? 
Anyway you could use https://github.com/openbmc/bmcweb/search?q=getDateTime ?

Reply:
Line 843: // YYYY-MM-DD format
I don't think getDateTime would be helpful in this case. getDateTime is std::time_t[in] std::string[out].
I'm trying to go from string to time_t. In hindsight I shouldn't have pass the date as a string over dbus (potential improvement later).